sale_import_base/components/importer_sale_channel.py

```diff
@@ -0,0 +1,202 @@
+        return new_sale_order
+
+    def _prepare_vals(self, data):
+        partner = self._process_partner(data)
+        address_invoice, address_shipping = self._process_addresses(
+            partner, data["address_invoicing"], data["address_shipping"]
+        )
+        so_vals = {
+            "partner_id": partner.id,
+            "partner_invoice_id": address_invoice.id,
+            "partner_shipping_id": address_shipping.id,
+            "si_amount_total": data["amount"]["amount_total"],
+            "si_amount_untaxed": data["amount"]["amount_untaxed"],
+            "si_amount_tax": data["amount"]["amount_tax"],
+            "order_line": [
+                (0, 0, self._prepare_sale_line(line)) for line in data["lines"]
+            ],
+            "sale_channel_id": self.collection.record_id,
+        }
+        result = self._execute_onchanges(so_vals)
+        return result
+
+    def _process_partner(self, data):
+        partner = self._find_partner(data)
+        vals = self._extract_address_info(data["address_customer"])
+        if partner:
+            partner.write(vals)
+            result = partner
+        else:
+            result = self.env["res.partner"].create(vals)
+            self.env["sale.channel.partner"].create(
+                {
+                    "external_id": data["address_customer"].get("external_id")
+                    or result.email,
+                    "sale_channel_id": self.collection.record_id,
+                    "partner_id": result.id,
+                }
+            )
+        return result
+
+    def _find_partner(self, data):
+        external_id = data["address_customer"].get("external_id")
+        binding = self.env["sale.channel.partner"].search(
+            [
+                ("external_id", "=", external_id),
+                ("sale_channel_id", "=", self.collection.record_id),
+            ]
+        )
+        sale_channel = self.env["sale.channel"].browse(self.collection.record_id)
+        if binding:
+            return binding.partner_id
+        elif sale_channel.allow_match_on_email:
+            partner = self.env["res.partner"].search(
+                [("email", "=", data["address_customer"]["email"])]
+            )
+            if partner:
+                self.env["sale.channel.partner"].create(
+                    {
+                        "external_id": partner.email,
+                        "sale_channel_id": self.collection.record_id,
+                        "partner_id": partner.id,
+                    }
+                )
+                return partner
+
+    def _extract_address_info(self, data):
+        result = {
+            "name": data.get("name") or data.get("email") or data.get("city"),
+            "street": data.get("street"),
+            "street2": data.get("street2"),
+            "zip": data.get("zip"),
+            "city": data.get("city"),
+            "email": data.get("email"),
+        }
+        if data.get("state_code"):
```

Comment:
Name field is not necessary for shipping and invoicing addresses, so I just put either name, or email, or worst case the city. Does this seem OK to you or should I force the shipping/invoicing address names to be the same as the partner name?